components/api/src/main/java/com/hotels/styx/api/FullHttpResponse.java

```diff
@@ -43,7 +43,29 @@
 import static java.util.stream.Collectors.toList;
 
 /**
- * HTTP response with a fully aggregated/decoded body.
+ * An immutable HTTP response object containing a finite size content body in full.
+ * <p>
+ * A {@code FullHttpResponse} is useful for HTTP response messages with
+ * finite body content such as returning a REST object.
```

Comment:
"A {@code FullHttpResponse} is useful for HTTP response messages with finite body content such as a REST object." perhaps?

The act of returning a message is not a part of a message.